tests/unit/unit_test_tables.py

```diff
@@ -902,3 +901,66 @@ def test_iter_metadata__no_etag(self):
             assert_equals(2, len(metadata))
             assert_equals((1,2, None), metadata[0])
             assert_equals((5, 1, None), metadata[1])
+
+    def test_iter_no_row_metadata(self):
+        data = "col1,col2\n" \
+               "1,2\n" \
+               "2,1\n"
+        cols = as_table_columns(StringIOContextManager(data))
+        headers = [SelectColumn.from_column(col) for col in cols]
+        with patch.object(io, "open", return_value=StringIOContextManager(data)):
+            table = CsvFileTable("syn123", "/fake/file/path", headers=headers)
+            for row in table:
+                assert_equals(2, len(row))
+
+    def test_iter_with_table_row_metadata(self):
+        data = "ROW_ID,ROW_VERSION,col\n" \
+               "1,2,\"I like trains\"\n" \
+               "5,1,\"weeeeeeeeeeee\"\n"
+        cols = as_table_columns(StringIOContextManager(data))
+        headers = [SelectColumn.from_column(col) for col in cols]
+        with patch.object(io, "open", return_value=StringIOContextManager(data)):
+            table = CsvFileTable("syn123", "/fake/file/path", headers=headers)
+            for row in table:
+                assert_equals(1, len(row))
+
+    def test_iter_with_file_view_row_metadata(self):
+        data = "ROW_ID,ROW_VERSION,ROW_ETAG,col\n" \
+               "1,2,etag1,\"I like trains\"\n" \
+                "5,1,etag2,\"weeeeeeeeeeee\"\n"
+        cols = as_table_columns(StringIOContextManager(data))
+        headers = [SelectColumn.from_column(col) for col in cols]
+        with patch.object(io, "open", return_value=StringIOContextManager(data)):
+            table = CsvFileTable("syn123", "/fake/file/path", headers=headers)
+            for row in table:
+                assert_equals(1, len(row))
+
+    def test_iter_with_row_metadata_in_header(self):
+        data = "ROW_ID,ROW_VERSION,ROW_ETAG,col\n" \
+               "1,2,etag1,\"I like trains\"\n" \
+                "5,1,etag2,\"weeeeeeeeeeee\"\n"
+        cols = as_table_columns(StringIOContextManager(data))
+        headers = [SelectColumn(name="ROW_ID", columnType="STRING"),
+                   SelectColumn(name="ROW_VERSION", columnType="STRING")] + \
+                  [SelectColumn.from_column(col) for col in cols]
+        with patch.object(io, "open", return_value=StringIOContextManager(data)):
+            table = CsvFileTable("syn123", "/fake/file/path", headers=headers)
+            for row in table:
+                assert_equals(3, len(row))
```

Comment:
I ran this test through the debugger
This has weird values when iterating:
`row` has the value of:
`['2', 'etag1', 'I like trains']`
`['1', 'etag2', 'weeeeeeeeeeee']`
which means you cut off only the ROW_ID column when iterating.
this does not seem like intended behavior

Reply:
As an additional note, since you are testing the `__iter__`method, all of your tests should assert the actual values you expect to get back from `__iter__`  instead of just the result's length.

Reply:
Sounds good.

There is also the possible case of  csv not contain Row Metadata, headers contain Row Metadata
This should only happen if the user provides a csv without Row Metadata but the `table` has `self.includeRowIdAndRowVersion==True`